src/components/checkbox-filter-group/checkbox-filter-group.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { BaseComponent } from '@/templates/base-component';
+import { CheckboxInner } from '../checkbox-inner/checkbox-inner';
+import sprite from '../../assets/images/sprite.svg';
+import { FilterGroup } from '@/enums/filter-name';
+import { Product } from '@/interfaces/product';
+
+export type value = {
+  quantityTotal: number;
+  value: string;
+  name: FilterGroup;
+  // name: keyof Pick<Product, 'brandName' | 'category' | 'for'>;
+};
+export class CheckboxFilterGroup extends BaseComponent {
+  private toggle: BaseComponent<'button'>;
+  private checkboxFilter: BaseComponent<'ul'>;
+  constructor(
+    private props: {
+      title: string;
+      name: string;
+      valuesAll: value[];
+      valuesFiltered: value[];
+      currentFilters: string[];
+    },
+    private cbGetValues: (filterName: string) => string[],
+    private cbRenderProducts: (products: Product[]) => void,
+  ) {
+    super('div', { className: 'filter-block filter__multy-choice' });
+    this.toggle = new BaseComponent('button', {
+      className: 'filter-block__toggle btn-reset',
+      type: 'button',
+      innerHTML: this.setToggleContent(),
+    });
+    this.toggle.getNode().addEventListener('click', () => {
+      this.toggleClass('close');
+    });
+    this.checkboxFilter = new BaseComponent('ul', {
+      className: 'checkbox-filter list-reset',
+    });
+
+    this.props.valuesAll.forEach((data) => {
```

Comment:
`valuesAll` is a bad name , it tells nothing about types, about entities or logic.